proxy/src/main/java/com/velocitypowered/proxy/protocol/StateRegistry.java

```diff
@@ -282,29 +283,46 @@ public boolean equals(Object o) {
             if (o == null || getClass() != o.getClass()) return false;
             PacketMapping that = (PacketMapping) o;
             return id == that.id &&
-                    protocolVersion == that.protocolVersion;
+                    protocolVersion == that.protocolVersion &&
+                    packetDecoding == that.packetDecoding;
         }
 
         @Override
         public int hashCode() {
-            return Objects.hash(id, protocolVersion);
+            return Objects.hash(id, protocolVersion, packetDecoding);
         }
     }
 
-    private static PacketMapping map(int id, int version, boolean handleFromBackend) {
-        return new PacketMapping(id, version, handleFromBackend);
+    private static PacketMapping map(int id, int version, boolean packetDecoding) {
+        return new PacketMapping(id, version, packetDecoding);
     }
     
-    private static PacketMapping map(int id, int version) {
+    /**
+     * Create a PacketMapping where packet decoding is <b>enabled</b>
+     * @param id Packet id
+     * @param version Protocol version
+     * @return PacketMapping with packet decoding enabled
+     */
+    private static PacketMapping map1(int id, int version) {
```

Comment:
As @MylesIsCool said, most packets Velocity handles are not decoded by default, so it would make sense to have a `map(int id, int version)` map that called `map(int id, int version, false)` (where `false` is `packetDecoding`).